File: go/82.go
package main

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates2(head *ListNode) *ListNode {
	res := &ListNode{0, head}
	pre := res
	for pre != nil && pre.Next != nil && pre.Next.Next != nil {
		nxt, nxxt := pre.Next, pre.Next.Next
		if nxt.Val != nxxt.Val {
			pre = pre.Next
		} else {
			for nxxt.Next != nil && nxxt.Val == nxxt.Next.Val {
				nxxt = nxxt.Next
			}
			pre.Next = nxxt.Next
		}
	}
	return res.Next
}

File: python/82.py
# Definition for singly-linked list.
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        res = ListNode(0, head)
        pre = res
        while pre and pre.next and pre.next.next:
            nxt, nxxt = pre.next, pre.next.next
            if nxt.val != nxxt.val:
                pre = pre.next
            else:
                while nxxt.next and nxxt.val == nxxt.next.val:
                    nxxt = nxxt.next
                pre.next = nxxt.next
        return res.next